src/app/scenes/Category.tsx

```diff
@@ -1,56 +1,60 @@
-import { BookState } from 'app/services/book';
-import {
-  Category as CategoryState,
-  CategoryCollectionState,
-  Actions as categoryActions,
-} from 'app/services/category';
+import { Actions as categoryActions } from 'app/services/category';
 import { getIdFromLocationSearch, isValidNumber } from 'app/services/category/utils';
 import { RidiSelectState } from 'app/store';
 import { Pagination } from 'app/components/Pagination';
-import { getIsIosInApp } from 'app/services/environment/selectors';
 import { getPageQuery } from 'app/services/routing/selectors';
 import { getIsMobile } from 'app/services/commonUI/selectors';
 
-interface CategoryStateProps {
-  isIosInApp: boolean;
-  isCategoryListFetched: boolean;
-  categoryList: CategoryState[];
-  categoryId: number;
-  category: CategoryCollectionState;
-  books: BookState;
-  page: number;
-  isMobile: boolean;
-}
-
-type Props = CategoryStateProps & ReturnType<typeof mapDispatchToProps>;
-
-interface State {
-  isInitialized: boolean;
-}
-
-class Category extends React.Component<Props, State> {
-  private initialDispatchTimeout?: number | null;
-
-  public state: State = {
-    isInitialized: false,
-  };
+const Category: React.FunctionComponent = () => {
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const location = useLocation();
+
+  const isCategoryListFetched = useSelector((state: RidiSelectState) => state.categories.isFetched);
+  const categoryList = useSelector((state: RidiSelectState) => state.categories.itemList) || [];
+  const categoryId = useSelector((state: RidiSelectState) =>
+    Number(getIdFromLocationSearch(state.router.location.search)),
+  );
+  const category = useSelector(
+    (state: RidiSelectState) =>
+      state.categoriesById[Number(getIdFromLocationSearch(state.router.location.search))],
+  );
+  const books = useSelector((state: RidiSelectState) => state.booksById);
+  const page = useSelector(getPageQuery);
+  const isMobile = useSelector(getIsMobile);
+
+  const isValidCategoryId = isValidNumber(categoryId);
+  const itemCount = category?.itemCount;
+  const ItemCountPerPage = 24;
+
+  const isFetched = () =>
+    category && category.itemListByPage[page] && category.itemListByPage[page].isFetched;
+
+  React.useEffect(() => {
+    dispatch(
+      categoryActions.initializeCategoriesWhole({
+        shouldFetchCategoryList: !isCategoryListFetched,
+        shouldInitializeCategoryId: !isValidCategoryId,
+      }),
+    );
+  }, []);
 
-  private isFetched = (page: number) => {
-    const { category } = this.props;
-    return category && category.itemListByPage[page] && category.itemListByPage[page].isFetched;
-  };
+  React.useEffect(() => {
+    if (isValidCategoryId) {
+      dispatch(categoryActions.cacheCategoryId({ categoryId }));
+      !isFetched() && dispatch(categoryActions.loadCategoryBooksRequest({ categoryId, page }));
+    }
+  }, [location]);
```

Comment:
```suggestion
  }, [categoryId]);
```
이렇게 변경하는게 명확하지 않을까요?

Reply:
넵 + `page` 도 추가할게요.